tracecontext/src/main/java/brave/propagation/tracecontext/TracestateFormat.java

```diff
@@ -107,9 +115,30 @@ boolean validateKey(CharSequence buffer, int beginKey, int endKey) {
     return true;
   }
 
+  /**
+   * https://www.w3.org/TR/trace-context-1 has some ambiguity about how to treat the value when
+   * considering whitespace. https://github.com/w3c/trace-context/pull/411 clarifies initial spaces
+   * are a part of the value. However, the value must end in at least one character in the range
+   * ' ' to '~', except ',' and '='. This implementation is based on the updated interpretation.
+   *
+   * <p>For example, pull 411 clarifies "" is not valid, and "   a" is a different value than " a".
+   */
   boolean validateValue(CharSequence buffer, int beginValue, int endValue) {
```

Comment:
Optional nit, seems a bit mysterious that only these private methods take CharSequence instead of String

Reply:
it was the narrowest interface thing. also, it is kindof a TODO to make the whole thing work with char buffers. want me to change this, or mostly just voicing the cognitive glitch?

Reply:
Just voicing no worries

Reply:
anyway it is good.. maybe I can find a way to juggle positions in a later PR. thanks for talking about it!